.github/workflows/extbuild.yml

```diff
@@ -74,28 +48,27 @@ jobs:
           parallelio_version: ${{ env.ParallelIO_VERSION }}
           enable_fortran: True
           install_prefix: $HOME/pio
-      - name: Cache ESMF
-        id: cache-esmf
-        uses: actions/cache@v3
-        with:
-          path: ~/ESMF
-          key: ${{ runner.os }}-${{ env.ESMF_VERSION }}-ESMF2
-      - name: Build ESMF
-        if: steps.cache-esmf.outputs.cache-hit != 'true'
-        uses: ./.github/actions/buildesmf
+      - name: Install ESMF
+        uses: esmf-org/install-esmf-action@v1
+        env:
+          ESMF_COMPILER: gfortran
+          ESMF_BOPT: g
+          ESMF_COMM: openmpi
+          ESMF_NETCDF: nc-config
+          ESMF_PNETCDF: pnetcdf-config
+          ESMF_INSTALL_PREFIX: /home/runner/ESMF
+          ESMF_PIO: external
+          ESMF_PIO_INCLUDE: /home/runner/pio/include
+          ESMF_PIO_LIBPATH: /home/runner/pio/lib
         with:
-          esmf_version: ${{ env.ESMF_VERSION }}
-          esmf_bopt: g
-          esmf_comm: openmpi
-          install_prefix: $HOME/ESMF
-          netcdf_c_path: /usr
-          netcdf_fortran_path: /usr
-          pnetcdf_path: /usr
-          parallelio_path: $HOME/pio
+          version: ${{ env.ESMF_VERSION }}
+          esmpy: false
+          cache: true
+      
       - name: Build CDEPS
         uses: ./.github/actions/buildcdeps
         with:
-          esmfmkfile: $HOME/ESMF/lib/libg/Linux.gfortran.64.openmpi.default/esmf.mk
+          esmfmkfile: $HOME/ESMF/lib/esmf.mk
```

Comment:
`install-esmf-action` set ESMFMKFILE for you, my recommendation if you didn't want to remove this from the `buildcdeps` action is to change this line to `esmfmkfile: $ESMFMKFILE` or `esmfmkfile: ${{env.ESMFMKFILE}}`

Reply:
done